src/iuse.cpp

```diff
@@ -3234,7 +3244,7 @@ int iuse::solder_weld( player *p, item *it, bool, const tripoint& )
     }};
 
     int pos = g->inv_for_filter( _("Repair what?"), [it]( const item &itm ) {
-        return itm.made_of_any( materials ) && !itm.is_ammo() && !itm.is_gun() && &itm != it;
+        return itm.made_of_any( materials ) && !itm.is_ammo() && is_firearm(itm) && &itm != it;
```

Comment:
Other places in this PR replace `is_gun` with `is_firearm`. In this line, it's replaced as well, but the `!` is dropped, too. This inverts the logic. Again: is this intended?